### PR TITLE
Update job request

### PR DESCRIPTION
**Pull request by wslulciuc**
This PR makes the following changes to `CreateJobRequest`:

- Renamed to `JobRequest`
- Explicitly use `@Getter`, `@EqualsAndHashCode` and `@ToString`
- Annotate `JobRequest. description` with `@Nullable`
- Annotate construct with `@JsonCreator` to allow for jackson deserialization
- Add `JobRequest.getDescription()` returning an optional string

**Related issue:** #193

### CODE REVIEW

File: src/test/java/marquez/api/JobResourceTest.java [hunk 1]
```diff
@@ -333,12 +333,12 @@ private Response getJob(String jobName) {
   }
 
   private Response insertJob(Job job) {
-    CreateJobRequest createJobRequest =
-        new CreateJobRequest(
-            job.getLocation(),
-            job.getDescription(),
+    JobRequest createJobRequest =
```
Comment (roaraya8) [inline src/test/java/marquez/api/JobResourceTest.java]: maybe renaming 'createJobRequest' to 'jobRequest' for consistency

File: src/test/java/marquez/api/JobIntegrationTest.java [hunk 2]
```diff
@@ -255,12 +255,12 @@ private void evaluateResponse(Response res, Job inputJob) {
...
 
   private Response createJobOnNamespace(String namespace, Job job) {
-    CreateJobRequest createJobRequest =
-        new CreateJobRequest(
-            job.getLocation(),
-            job.getDescription(),
+    JobRequest createJobRequest =
+        new JobRequest(
             job.getInputDataSetUrns(),
-            job.getOutputDataSetUrns());
```
Comment (roaraya8) [inline src/test/java/marquez/api/JobIntegrationTest.java]: I guess the plan would be to later normalize  '..DataSet..' to '..Dataset..' everywhere, also for consistency

Comment (wslulciuc) [reply src/test/java/marquez/api/JobIntegrationTest.java]: Yep, exactly! Mind opening an issue to capture the change?

File: src/test/java/marquez/api/JobResourceTest.java (hunk 1 above)
Comment (wslulciuc) [reply src/test/java/marquez/api/JobResourceTest.java]: Good catch!

**Pull request merged**